### DISCUSSION TITLE
custom admin urls routing wrong?

### DISCUSSION THREAD

**Discussion by stygmate** [jazzband/django-polymorphic#804] [Q&A] [answered]
in one of my child model i have a change password custom urls

```python
    def get_urls(self):
        return [
                   path(
                       '<path:object_id>/password/',
                       self.admin_site.admin_view(self.mymodel_change_password),
                       name='mymodel_password_change',
                   ),
               ] + super().get_urls()
```

and when i try to access:
```
/admin/importers/importateur/1/password/
```
the url become:
```
/admin/importers/importateur/1/password/change/
```
and 1/password is catched as object id for the change view

Any help ?

**Comment by stygmate** [comment]
anyone ?

**Comment by JohananOppongAmoateng** [comment, marked as answer]
The issue is caused by using `<path:object_id>`, which is greedy and captures `1/password` as the object ID, so Django routes it to the change view.

**Fix:** don’t use `<path:object_id>` in admin URLs. Use a non-greedy converter instead.

```python
def get_urls(self):
    return [
        path(
            '<int:object_id>/password/',
            self.admin_site.admin_view(self.mymodel_change_password),
            name='mymodel_password_change',
        ),
    ] + super().get_urls()
```

This prevents `password` from being swallowed by the change view and resolves the URL correctly.

**Comment by JohananOppongAmoateng** [comment]
@bckohan I think we can convert this into a discussion
